include/openssl/bn.h

```diff
@@ -136,14 +136,7 @@ int BN_is_word(const BIGNUM *a, const BN_ULONG w);
 int BN_is_odd(const BIGNUM *a);
 
 # define BN_one(a)       (BN_set_word((a),1))
-
-void BN_zero_ex(BIGNUM *a);
-
-# if OPENSSL_API_COMPAT >= 0x00908000L
-#  define BN_zero(a)      BN_zero_ex(a)
-# else
-#  define BN_zero(a)      (BN_set_word((a),0))
-# endif
+# define BN_zero(a)      (BN_set_word((a),0))
```

Comment:
> Look at the code.  It was done wrong.  This fixes it.

How is that done wrong?  `BN_zero_ex` sets `top` and `neg` to 0, which is exactly the criteria for a non-negative zero (have a look at the `BN_is_*` functions to see that they all check at least on of these fields).

If you want to keep just one define, it should be this:

``` C
# define BN_zero(a)    (BN_zero_ex(a))
```